client/pages/stops-list.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+
+import MapList from './components/list-map';
+
+export default class TripsList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onMouseEnterHandler = this.onMouseEnterHandler.bind(this);
+    this.onMouseLeaveHandler = this.onMouseLeaveHandler.bind(this);
+    this.state = {
+      stopsList: [],
+      hover: false,
+      mapCoords: {}
+    };
+    fetch('/api/stopslist')
+      .then(res => res.json())
+      .then(result => {
+
+        this.setState({ stopsList: result });
+      })
+      .catch(err => {
+        console.error(err);
+      });
+  }
+
+  onMouseEnterHandler(event) {
+    this.setState({ hover: true });
+    this.state.stopsList.map(stop => {
+      if (String(stop.stopId) === event.currentTarget.id) {
+        // console.log(((new Date(stop.beginTime).getTime() - new Date(stop.endTime).getTime()) / 1000) / 60);
+        return this.setState({ mapCoords: { lat: Number(stop.stopLocation.lat), lng: Number(stop.stopLocation.lng) } });
+      }
+      return null;
+    });
```

Comment:
Rather than use `.map()` just for looping purposes, I recommend breaking this into two steps.
1. find the matching element with Array.prototype.find()
2. update your state

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#when_not_to_use_map

Reply:
Update made